lib/elastic_apm/central_config.rb

```diff
@@ -149,9 +153,10 @@ def headers
       { 'Etag': @etag }
     end
 
-    def schedule_next_fetch(resp)
+    def schedule_next_fetch(resp=nil)
+      headers = resp ? resp.headers : {}
       seconds =
-        if (cache_header = resp.headers['Cache-Control'])
+        if (cache_header = headers['Cache-Control'])
```

Comment:
If you use the implementation I suggested, `resp` will either be a real response or `nil`, so this will be enough:

```suggestion
        if (cache_header = resp&.headers['Cache-Control'])
```

I think...